airbyte-server/src/main/java/io/airbyte/server/apis/SourceApiController.java

```diff
@@ -58,6 +58,7 @@ public CheckConnectionRead checkConnectionToSource(final SourceIdRequestBody sou
   @Post("/check_connection_for_update")
   @Secured({EDITOR})
   @SecuredWorkspace
+  @ExecuteOn(TaskExecutors.IO)
```

Comment:
thinking out loud: I think it'll be very useful to have a spot explaining what this does and was going to suggest in code comments. However, clicking into the annotation provides enough reader information. Perhaps the internal micronaut guide?